### PR TITLE
fix: Create job from preset test does not have enough time to complete (#4617)

### LINKED ISSUE #4617: com.example.transcoder.CreateJobFromTemplateTest: test_CreateJobFromTemplate failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: d5e050261047c50dd502555b4dedfe877169b49c
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/c61dd333-6269-4a0e-b6af-536e31d3051e), [Sponge](http://sponge2/c61dd333-6269-4a0e-b6af-536e31d3051e)
status: failed
<details><summary>Test output</summary><br><pre>java.lang.AssertionError: 

Expected: a string containing "SUCCEEDED"
     but: was "Job state: RUNNING
"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at com.example.transcoder.CreateJobFromTemplateTest.test_CreateJobFromTemplate(CreateJobFromTemplateTest.java:136)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
</pre></details>

### PR DESCRIPTION
**Pull request by irataxy**
Fixes #4617

> It's a good idea to open an issue first for discussion.

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [X] Appropriate changes to README are included in PR
- [X] API's need to be enabled to test (tell us)
- [X] Environment Variables need to be set (ask us to set them)
- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [X] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: media/transcoder/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  </properties>
+  <!-- [START dependencies] -->
+  <!--  Using libraries-bom to manage versions.
+  See https://github.com/GoogleCloudPlatform/cloud-opensource-java/wiki/The-Google-Cloud-Platform-Libraries-BOM -->
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>libraries-bom</artifactId>
+        <version>15.0.0</version>
```
Comment (lesv) [inline media/transcoder/pom.xml:46]: 16.3.0

File: media/transcoder/pom.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+  <dependencies>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-video-transcoder</artifactId>
+      <version>0.1.3</version>
```
Comment (lesv) [inline media/transcoder/pom.xml:56]: 0.1.4

File: media/transcoder/pom.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.13.1</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-storage</artifactId>
+      <version>1.113.3</version>
```
Comment (lesv) [inline media/transcoder/pom.xml:79]: 1.113.9

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      }
+      bucket.delete();
+    }
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    PROJECT_ID = requireEnvVar("GOOGLE_CLOUD_PROJECT");
+    PROJECT_NUMBER = requireEnvVar("GOOGLE_CLOUD_PROJECT_NUMBER");
```
Comment (lesv) [inline media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:88]: Let's change this to TRANSCODER_PROJECT_NUMBER

File: .kokoro/tests/run_tests.sh [hunk 5]
```diff
@@ -55,6 +55,7 @@ if [[ "$SCRIPT_DEBUG" != "true" ]]; then
 
     # Setup required env variables
     export GOOGLE_CLOUD_PROJECT=java-docs-samples-testing
+    export GOOGLE_CLOUD_PROJECT_NUMBER="779844219229" # For Transcoder samples
```
Comment (lesv) [inline .kokoro/tests/run_tests.sh:58]: Let's change this to TRANSCODER_PROJECT_NUMBER

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    Bucket bucket = storage.get(bucketName);
+    if (bucket != null) {
+      Page<Blob> blobs = bucket.list();
+
+      for (Blob blob : blobs.iterateAll()) {
+        System.out.println(blob.getName());
+        storage.delete(bucketName, blob.getName());
+      }
+      bucket.delete();
+    }
```
Comment (lesv) [inline media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:81]: Why are you deleting the blobs if you are also deleting the bucket?

**Review by lesv — CHANGES_REQUESTED** [review]
PTAL - I stopped on the first test, @shubha-rajan should do a full review including all the tests.

From me:  I'd like to rename the EnvVar so it's clear it's not our standard ProjectNumber

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java (hunk 6 above)
Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:81]: I had patterned these tests after the python code samples which required deleting objects in the bucket first.

File: .kokoro/tests/run_tests.sh (hunk 5 above)
Comment (irataxy) [reply .kokoro/tests/run_tests.sh]: Done

File: media/transcoder/pom.xml (hunk 1 above)
Comment (irataxy) [reply media/transcoder/pom.xml]: Done

File: media/transcoder/pom.xml (hunk 2 above)
Comment (irataxy) [reply media/transcoder/pom.xml]: Done

File: media/transcoder/pom.xml (hunk 3 above)
Comment (irataxy) [reply media/transcoder/pom.xml]: Done

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java (hunk 4 above)
Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java]: Done for all tests

**Comment by lesv** [comment]
@shubha-rajan Should review

@irataxy It looks like your RegionTags (see snippet-bot reporting) aren't correct, also it looks like your latest commit broke a bunch of things.

**Comment by shubha-rajan** [comment]
> @shubha-rajan Should review

I've set aside some time to review this in the afternoon.

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java (hunk 6 above)
Comment (lesv) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:81]: You don't need to delete the blob's, it just costs more that way.

Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:81]: When I remove the blob deletion and run this locally, I get the following errors:

com.google.cloud.storage.StorageException: The bucket you tried to delete was not empty.

Comment (lesv) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:81]: Ok, keep it in then --  I've deleted buckets in the past w/ artifacts in them.

**Comment by lesv** [comment]
You still have conflicts that you need to resolve.

**Review by irataxy — COMMENTED** [review]
Reviewed

**Comment by lesv** [comment]
@shubha-rajan This looks ready for rereview - ping me once you are done and I'll review.

File: .kokoro/tests/run_tests.sh [hunk 7]
```diff
@@ -55,7 +55,7 @@ if [[ "$SCRIPT_DEBUG" != "true" ]]; then
 
     # Setup required env variables
     export GOOGLE_CLOUD_PROJECT=java-docs-samples-testing
-    export GOOGLE_CLOUD_PROJECT_NUMBER="779844219229" # For Transcoder samples
+    export TRANSCODER_PROJECT_NUMBER="779844219229" # For Transcoder samples
```
Comment (shubha-rajan) [inline .kokoro/tests/run_tests.sh:58]: Should this be hard coded into the script, or should we source it from the secrets file on Cloud Devrel Kokoro Resources?

**Review by shubha-rajan — APPROVED** [review]
@irataxy Thanks for fixing those merge conflicts! I was a bit confused as to why you were adding new samples when the PR title implied that you intended to fix timeouts. This makes more sense.

This looks good. I just have one question about hardcoding the project number environment variable vs. sourcing it.

File: .kokoro/tests/run_tests.sh (hunk 7 above)
Comment (irataxy) [reply .kokoro/tests/run_tests.sh:58]: Yes, this PR is just for updating the test durations due to flakiness.

I'm open to sourcing these if that is the best practice. If so, can you point me to an example?

Comment (lesv) [reply .kokoro/tests/run_tests.sh:58]: It's fine either way.  Different repo's do different things.  You'll note that the line above is also hardcoded.

**Review by lesv — APPROVED** [review]

**Pull request merged**